python/pyspark/sql/tests.py

```diff
@@ -558,6 +558,18 @@ def check_datatype(datatype):
         _verify_type(PythonOnlyPoint(1.0, 2.0), PythonOnlyUDT())
         self.assertRaises(ValueError, lambda: _verify_type([1.0, 2.0], PythonOnlyUDT()))
 
+        schema = StructType().add("key", LongType()).add("val", PythonOnlyUDT())
+        df = self.spark.createDataFrame(
+            [(i % 3, PythonOnlyPoint(float(i), float(i))) for i in range(10)],
+            schema=schema)
+        df.show()
```

Comment:
`DataFrame.show()` gives unnecessary stringification, so this test ends up testing unnecessary stuff (in fact it would fail if the UDT didn't have `__str__`. I would use `collect()` to force materialization instead.